devicetypes/shinasys/sihas-multipurpose-sensor.src/sihas-multipurpose-sensor.groovy

```diff
@@ -0,0 +1,312 @@
+            state("temperature", label: '${currentValue}°', unit: "C",
+                    backgroundColors: [
+                            [value: 31, color: "#153591"],
+                            [value: 44, color: "#1e9cbb"],
+                            [value: 59, color: "#90d2a7"],
+                            [value: 74, color: "#44b621"],
+                            [value: 84, color: "#f1d801"],
+                            [value: 95, color: "#d04e00"],
+                            [value: 96, color: "#bc2323"]
+                    ]
+            )
+        }
+        valueTile("humidity", "device.humidity", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+            state "humidity", label: '${currentValue}% humidity', unit: ""
+        }
+        
+        valueTile("illuminance", "device.illuminance", width: 2, height: 2) {
+            state "luminosity", label:'${currentValue} ${unit}', unit:"lux"
+        }
+        
+        valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+            state "battery", label: '${currentValue}% battery', unit: ""
+        }
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+        }
+    
+        main(["motion", "temperature"])
+        details(["motion", "temperature", "humidity", "illuminance", "battery", "refresh"])
+    }
+}
+
+private List<Map> collectAttributes(Map descMap) {
+    List<Map> descMaps = new ArrayList<Map>()
+
+    descMaps.add(descMap)
+
+    if (descMap.additionalAttrs) {
+        descMaps.addAll(descMap.additionalAttrs)
+    }
+
+    return  descMaps
+}
+
+def parse(String description) {
+    log.debug "description: $description"
+    Map map = zigbee.getEvent(description)
+    //log.debug "map: $map"
+    if (!map) {
+        if (description?.startsWith('zone status')) {
+            map = parseIasMessage(description)
+        } else if (description?.startsWith('read attr')) {
+            Map descMap = zigbee.parseDescriptionAsMap(description)
+
+            if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap.value) {
+                //log.info "BATT METRICS - attr: ${descMap?.attrInt}, value: ${descMap?.value}, decValue: ${Integer.parseInt(descMap.value, 16)}, currPercent: ${device.currentState("battery")?.value}, device: ${device.getDataValue("manufacturer")} ${device.getDataValue("model")}"
+                List<Map> descMaps = collectAttributes(descMap)
+                def battMap = descMaps.find { it.attrInt == 0x0020 }
+
+                if (battMap) {
+                    map = getBatteryResult(Integer.parseInt(battMap.value, 16))
+                }
+            } else if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002 && descMap.commandInt != 0x07) {
```

Comment:
change `0x0500` to `zigbee.IAS_ZONE_CLUSTER` and `0x0002` to `zigbee.ATTRIBUTE_IAS_ZONE_STATUS`